osf/models/spam.py

```diff
@@ -266,3 +280,33 @@ def do_check_spam(self, author, author_email, content, request_headers, update=T
                 self.spam_data['who_flagged'] = 'oopspam'
                 self.spam_data['oopspam_data'] = oopspam_details
         return akismet_is_spam or oopspam_is_spam
+
+    def moderate_domains(self, content, confirm_spam=False):
+        domains = re.findall(
```

Comment:
Wrap this in a set, please (ideally using groups and `finditer` as shown in the comment about the regex)